docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
+    {
+        if ((a == double.MaxValue && b > 0) || (b == double.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+}
+```
+
+### &lt;typeparam&gt;
+
+The `<typeparam>` tag functions just like the `<param>` tag but for generic type or method declarations to describe a generic parameter.
+Since our math library doesn't have a generic method I'll just bring in `SomeClass` to demonstrate it.
+
+```csharp
+/// <summary>
+/// This class does something.
+/// </summary>
+public class SomeClass
+{
+    /// <summary>
+    /// Does generic stuff.
+    /// </summary>
+    /// <typeparam name="T">The generic stuff to do.</typeparam>
+    public void DoGenericStuff<T>()
+    {
+
+    }
+}
+```
+
+### &lt;paramref&gt;
+
+Sometimes you might be in the middle of describing what a method does in what could be a `<summary>` tag and you might want to make a reference
+to a parameter, the `<paramref>` tag is great for just this. Let's update the summary of our double based `Add` method. Like the `<param>` tag
+the parameter name is specified in the **required** `name` attribute.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// <para>The main <c>Math</c> class.</para>
+/// <para>Contains all methods for performing basic math functions.</para>
+/// </summary>
+public class Math
+{
+    /// <summary>
+    /// Adds two doubles <paramref name="a"/> and <paramref name="b"/> and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two doubles.
+    /// </returns>
+    /// <exception cref="System.OverflowException">Thrown when one parameter is max and the other is greater than zero.</exception>
+    /// See <see cref="Math.Add(int, int)"/> to add integers.
+    /// <param name="a">A double precision number.</param>
+    /// <param name="b">A double precision number.</param>
+    public static double Add(double a, double b)
+    {
+        if ((a == double.MaxValue && b > 0) || (b == double.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+}
+```
+
+### &lt;typeparamref&gt;
+
+The `<typeparamref>` tag functions just like the `<paramref>` tag but for generic type or method declarations to describe a generic parameter.
+Since our math library doesn't have a generic method I'll just bring in `SomeClass` to demonstrate it.
+
+```csharp
+/// <summary>
+/// This class does something.
+/// </summary>
+public class SomeClass
+{
+    /// <summary>
+    /// Does generic stuff <typeparamref name="T"/>.
+    /// </summary>
+    /// <typeparam name="T">The generic stuff to do.</typeparam>
+    public void DoGenericStuff<T>()
+    {
+
+    }
+}
+```
+
+### &lt;list&gt;
+
+Although we don't get to use it with our math library the `<list>` tag is a very useful tag because it lets you format documentation information as an ordered list, unordered list or table.
```

Comment:
You could add a list of the numeric types supported by your math library in the <remarks> section of the class.